osu.Game.Rulesets.Catch/UI/CatcherArea.cs

```diff
@@ -155,11 +155,21 @@ private void load()
                         Anchor = Anchor.TopCentre,
                         Origin = Anchor.BottomCentre,
                     },
-                    createCatcherSprite().With(c =>
-                    {
-                        c.Anchor = Anchor.TopCentre;
-                    })
                 };
+
+                updateCatcher();
+            }
+
+            private Drawable catcherSprite;
+
+            private void updateCatcher()
+            {
+                catcherSprite?.Expire();
+
+                Add(catcherSprite = createCatcherSprite().With(c =>
+                {
+                    c.Anchor = Anchor.TopCentre;
+                }));
```

Comment:
Related to the comment above: maybe worth considering caching the three sprites and cycling the cached instances? It might be a bit of an issue with animated sprites, though (to make sure animation cycle is restarted upon switch).

Reply:
I think this is fine for now...